ledger/src/blockstore_db.rs

```diff
@@ -996,15 +997,31 @@ impl Default for ShredStorageType {
     }
 }
 
+/// Advanced options for blockstore.
+/// The each advanced option might also be used as a tag that supports
+/// group-by operation when reporting Blockstore metrics.
+#[derive(Clone)]
+pub struct BlockstoreAdvancedOptions {
+    // Determine how to store both data and coding shreds. Default: RocksLevel.
+    pub shred_storage_type: ShredStorageType,
+}
+
+impl Default for BlockstoreAdvancedOptions {
+    fn default() -> Self {
+        Self {
+            shred_storage_type: ShredStorageType::RocksLevel,
+        }
+    }
+}
+
```

Comment:
I'm not fully convinced we need to do this, but what do you think about breaking the addition of `BlockstoreAdvancedOptions` into its' own PR, and then having this PR update the metrics reporting?

Reply:
Yes, that's what I mentioned previously that I mixed two PRs into one.  One for introducing the BlockstoreAdvancedOptions, and one for the tag itself.

Reply:
So the way I am thinking about this is: for any field inside BlockstoreAdvancedOptions, we support group by.
For anything inside BlockstoreOptions but not in BlockstoreAdvancedOptions does not.

For instance, #23578, which is based on another two PRs, has three fields inside BlockstoreAdvancedOptions, and they all support the group by operation in metric look-up.

Reply:
Got it; I'm good leaving it as one PR